presto-main/src/test/java/com/facebook/presto/type/BenchmarkDecimalOperators.java

```diff
@@ -355,6 +397,11 @@ protected void generateProcessor(String expression)
             processor = new ExpressionCompiler(metadata).compilePageProcessor(rowExpression("true"), ImmutableList.of(rowExpression(expression))).get();
         }
 
+        protected void setDoubleMaxValue(double doubleMaxValue)
```

Comment:
Could you elaborate what do you have in mind?

`BaseState` provides some facility to generate random pages, so it is ok, that it provides some setters, to controls the behaviour of returned random values. I am not sure if I understood you correctly, but you would like `BaseState` to have ability to register custom random value generator for specific types and one benchmark, would overwrite a default generator for `DoubleType`. To be honest, I am not sure if this effort and complicating/increasing number of classes/abstractions is worth that in that case.